File: src/python/executor.py
from __future__ import print_function # needs to be first statement in file
import sys
import subprocess

arg_len = len(sys.argv)
if arg_len > 1:
    pos = int(sys.argv[1])
    langs = sys.argv[2]
else:
    sys.exit(1)


if pos == 0:
    print("H", end='')
elif pos == 1:
    print("e", end='')
elif pos == 2:
    print("l", end='')
elif pos == 3:
    print("l", end='')
elif pos == 4:
    print("o", end='')
elif pos == 5:
    print(" ", end='')
elif pos == 6:
    print("W", end='')
elif pos == 7:
    print("o", end='')
elif pos == 8:
    print("r", end='')
elif pos == 9:
    print("l", end='')
elif pos == 10:
    print("d", end='')
elif pos == 11:
    print("!", end='')
elif pos == 12:
    print("\n", end='')

sys.stdout.flush()


if pos >= 0 and pos < 12:
    if pos == 10 or pos == 11:
        target_lang = langs
        subprocess.call(["./controller.sh", str(target_lang), str(pos+1), str(langs)])
    else:
        target_lang, next_langs = langs.split(" ", 1)
        subprocess.call(["./controller.sh", str(target_lang), str(pos+1), str(next_langs)])

File: src/c/executor.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>


void partialPrint(int);
void notify(int, char*);

int main(int argc, char *argv[]) {
    char *langs;
    int pos = -1;

    if (argc > 1) {
        pos = atoi(argv[1]);
        langs = argv[2];
        if (pos > 12)
            return EXIT_FAILURE;
        partialPrint(pos);
        notify(pos, langs);
    }
    return 0;
}



void partialPrint(int pos) {
    if (pos == 0) {
        printf("%s", "H");
    } else if (pos == 1) {
        printf("%s", "e");
    } else if (pos == 2) {
        printf("%s", "l");
    } else if (pos == 3) {
        printf("%s", "l");
    } else if (pos == 4) {
        printf("%s", "o");
    } else if (pos == 5) {
        printf("%s", " ");
    } else if (pos == 6) {
        printf("%s", "W");
    } else if (pos == 7) {
        printf("%s", "o");
    } else if (pos == 8) {
        printf("%s", "r");
    } else if (pos == 9) {
        printf("%s", "l");
    } else if (pos == 10) {
        printf("%s", "d");
    } else if (pos == 11) {
        printf("%s", "!");
    } else if (pos == 12) {
        printf("%s", "\n");
    }

    fflush(stdout);
}

void notify(int pos, char* nlangs) {
    if (pos >= 0 || pos <= 11) {
        if (pos >= 10) {
            char cmd[100];
            char lLang[250];
            strcpy(lLang, "\'");
            strcat(lLang, nlangs);
            strcat(lLang, "\'");
            sprintf(cmd, "./controller.sh %s %d %s", lLang, pos+1, lLang);
            system(cmd);
        } else {
            // Get next target lang and execute logic.sh
            char *targetLang = strtok(nlangs, " ");
            char nextLangs[500];

            char *nlang = strtok(NULL, " ");
            int first = 0;
            while (nlang != NULL) {
                if (first == 0) {
                    strcpy(nextLangs, "\'");
                    strcat(nextLangs, nlang);
                    first++;
                } else {
                    strcat(nextLangs, " ");
                    strcat(nextLangs, nlang);
                }    

                nlang = strtok(NULL, " ");
            }
            strcat(nextLangs, "\'");
            char cmd[100];
            sprintf(cmd, "./controller.sh %s %d %s", targetLang, pos+1, nextLangs);
            system(cmd);
        }
    }

}

